modules/katamari/src/main/ts/ephox/katamari/api/Strings.ts

```diff
@@ -38,8 +39,12 @@ export const ensureTrailing = (str: string, suffix: string): string => {
   return endsWith(str, suffix) ? str : StrAppend.addToEnd(str, suffix);
 };
 
-export const contains = (str: string, substr: string): boolean => {
-  return str.indexOf(substr) !== -1;
+export const contains = (str: string, substr: string, start: number = 0, end?: number): boolean => {
+  if (Type.isUndefined(end) && start === 0) {
+    return str.indexOf(substr) !== -1;
+  } else {
+    return str.substring(start, end).indexOf(substr) !== -1;
```

Comment:
Couldn't this use the `position`-parameter of the `indexOf`-function, instead of generating new strings? Maybe javascript substring generation is more efficient than I give it credit for or the `position`-parameter introduces more complexity than it solves, but it feels like that might be faster.

Reply:
Good point! So I will resort to `substring` only when the `end` parameter is applied.

Reply:
This could actually be optimised further I believe. For example:
```ts
const idx = str.indexOf(substr, start);
if (idx !== -1) {
  return Type.isUndefined(end) ? true : idx + substr.length < end;
} else {
  return false;
}
```
This would safe the overhead of having to do the `substring` call all the time when `end` is passed since we can just compare if the matched index is within the bounds of the end.

Reply:
there is no limit to perfection...

Reply:
Indeed. I would have written that third line as `return Type.isUndefined(end) || idx + substr.length <= end;` 😂